### PR TITLE
fix: Use single-file-artifact-gen in Create an Artifact

### PR DESCRIPTION
**Pull request by lluiscampos**

### CODE REVIEW

File: 06.Artifact-creation/01.Create-an-Artifact/docs.md [hunk 1]
```diff
@@ -42,38 +42,45 @@ The remaining flags specify the parameters used to [match devices to deployments
...
+
 For example, assume that you want copy a new `authorized_keys` file to the `/home/${USER}/.ssh`
-directory on your devices, where `USER` holds your user name. We first store the path to the destination directory and file into two separate files, for packaging purposes:
+directory on your devices, where `USER` holds your user name.
 
-```bash
-echo /home/${USER}/.ssh > dest_dir # store the destination target directory
-echo authorized_keys > filename # store the filename of the file we want to update
-```
+For this example, we are creating a *module-image*, using the [single file](https://hub.mender.io/t/single-file/486/26?target=_blank) [Update Module](../../06.Artifact-creation/08.Create-a-custom-Update-Module/docs.md).
```
Comment (TheYoctoJester) [inline 06.Artifact-creation/01.Create-an-Artifact/docs.md:52]:
```suggestion
We will use the [single file](https://hub.mender.io/t/single-file/486/26?target=_blank) [Update Module](../../06.Artifact-creation/08.Create-a-custom-Update-Module/docs.md) to create a *module-image*.
```

File: 06.Artifact-creation/01.Create-an-Artifact/docs.md [hunk 2]
```diff
@@ -42,38 +42,45 @@ The remaining flags specify the parameters used to [match devices to deployments
...
-
-#### Update Modules generation script
+Note specifically that in this case we are creating a *module-image*, using the [single
+file](https://hub.mender.io/t/single-file/486/26?target=_blank) [Update
+Module](../../06.Artifact-creation/08.Create-a-custom-Update-Module/docs.md). The Artifact created
+will be compatible with the *raspberrypi4* device type, although you can specify multiple device
```
Comment (TheYoctoJester) [inline 06.Artifact-creation/01.Create-an-Artifact/docs.md:76]: can you clarify how? It is not clear if passing multiple `--device-type` flags or a comma separated list is the way to go.

File: 06.Artifact-creation/01.Create-an-Artifact/docs.md [hunk 3]
```diff
@@ -42,38 +42,45 @@ The remaining flags specify the parameters used to [match devices to deployments
...
+types if needed. The name of the Artifact is declared as *updated-authorized_keys-1.0*, we set the
+version of the software to *1.0* and indicate that it will be installed in the *rootfs* partition.
+The resulting file `artifact.mender` holds the Artifact.
 
 <!--AUTOVERSION: "mendersoftware/mender/blob/%/support"/mender-->
-You can see the above example in the [single file Update Module](https://hub.mender.io/t/single-file/486?target=_blank). You can also see how simple it is to [write a custom Update Module.](https://github.com/mendersoftware/mender/blob/master/support/modules/single-file?target=_blank)
+You can also see how simple it is to [write a custom Update Module.](https://github.com/mendersoftware/mender/blob/master/support/modules/single-file?target=_blank)
```
Comment (TheYoctoJester) [inline 06.Artifact-creation/01.Create-an-Artifact/docs.md:82]:
```suggestion
To get started writing your own custom one, please check out the [single file Update Module.](https://github.com/mendersoftware/mender/blob/master/support/modules/single-file?target=_blank)
```

**Review by TheYoctoJester — APPROVED** [review]
No major blockers, just some language.

File: 06.Artifact-creation/01.Create-an-Artifact/docs.md [hunk 4]
```diff
@@ -42,38 +42,45 @@ The remaining flags specify the parameters used to [match devices to deployments
 
 Creating an Artifact takes a different form in the case of [application updates](../../02.Overview/01.Introduction/docs.md#application-updates).
 
+For application updates, the command for `mender-artifact` tool will be different depending on the
+implementation details of the actual update module. Most of the Update Modules come with an script
```
Comment (tranchitella) [inline 06.Artifact-creation/01.Create-an-Artifact/docs.md:46]:
```suggestion
implementation details of the actual update module. Most of the Update Modules come with a script
```

File: 06.Artifact-creation/01.Create-an-Artifact/docs.md [hunk 5]
```diff
@@ -42,38 +42,45 @@ The remaining flags specify the parameters used to [match devices to deployments
...
+to simplify the Artifact creation so that the implementation details are hidden to the final user.
```
Comment (tranchitella) [inline 06.Artifact-creation/01.Create-an-Artifact/docs.md:47]:
```suggestion
to simplify the Artifact creation so that the implementation details are hidden from the final user.
```

File: 06.Artifact-creation/01.Create-an-Artifact/docs.md [hunk 6]
```diff
@@ -42,38 +42,45 @@ The remaining flags specify the parameters used to [match devices to deployments
...
+
 For example, assume that you want copy a new `authorized_keys` file to the `/home/${USER}/.ssh`
```
Comment (tranchitella) [inline 06.Artifact-creation/01.Create-an-Artifact/docs.md:49]:
```suggestion
For example, assume that you want to copy a new `authorized_keys` file to the `/home/${USER}/.ssh`
```

**Review by tranchitella — APPROVED** [review]
Looks good, just a few comments (on top of the others I also agree with).

File: 06.Artifact-creation/01.Create-an-Artifact/docs.md (hunk 3 above)
Comment (lluiscampos) [reply 06.Artifact-creation/01.Create-an-Artifact/docs.md:82]: "To get started" I would point the users to [the tutorial in Mender Docs](https://docs.mender.io/artifact-creation/create-a-custom-update-module) rather than the source code of a "simple" update module.

I'll see how can I rephrase this to have both :)

Comment (lluiscampos) [reply 06.Artifact-creation/01.Create-an-Artifact/docs.md:82]: Okey, this is actually mentioned below.

I will them modify the text here to:

```suggestion
Inspect the source code of the [single file Update Module](https://github.com/mendersoftware/mender/blob/master/support/modules/single-file?target=_blank) to learn about implementation details of this module.
```

File: 06.Artifact-creation/01.Create-an-Artifact/docs.md (hunk 2 above)
Comment (lluiscampos) [reply 06.Artifact-creation/01.Create-an-Artifact/docs.md:76]: I'll try
